server/src/controllers/ChapterUser/resolver.ts

```diff
@@ -195,6 +197,58 @@ export class ChapterUserResolver {
       throw Error('You cannot ban yourself');
     }
 
+    const userEvents = await prisma.event_users.findMany({
+      where: {
+        user_id: userId,
+        event: { chapter_id: chapterId },
+      },
+      include: {
+        event: {
+          include: { chapter: true, event_users: { include: { rsvp: true } } },
+        },
+        rsvp: true,
+      },
+    });
+    await prisma.event_users.deleteMany({
+      where: {
+        user_id: userId,
+        event: { chapter_id: chapterId },
+      },
+    });
```

Comment:
the tests aren't complaining, so this is working fine, but I am wondering how?

Because, reading the sequence.

- we are finding the EventUsers with rsvp
- we are deleting the event_user and their events
- we are filtering the deleted users and events, and finding their RSVP? (while they are deleted)
- we create an array of the filtered RSVP from the deleted events?
- we filter the deleted events' google calendar ids?
- we create an array of google calender ids?
- then we create nested array of both RSVP and google caalendar ids.
- then we return them

So, how can this connect, after we have deleted the events? 🤔

Reply:
Generally the "what happens here" question is valid.

- Get array with user EventUsers for events from banned chapter, including Event relation.
- Delete user EventUsers for events from banned chapter.
- Filter array to leave entry only if user is/was attending (`attendingEvents`).
- Update waitlist for events from `attendingEvents`, if capacity and waitlist allows for it.
- Filter `attendingEvents` to leave entry only if there's google `calendar_event_id` associated with the event (`eventsWithCalendar`).
- Update calendars attending lists for events from `eventsWithCalendar`.
- Create ban and return.

And yeah, we await for the initial array, before deleting entries in the database. Then we are working on the data in array.